Gems/Atom/RHI/DX12/Code/Source/RHI/ShaderUtils.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <Atom/RHI/PipelineStateDescriptor.h>
+#include <Atom/RHI.Reflect/DX12/ShaderStageFunction.h>
+
+namespace AZ::DX12
+{
+    namespace ShaderUtils
+    {
+        //! Patch a shader bytecode with the proper values of the specialization constants found in the
+        //! pipeline descriptor.
```

Comment:
Can you add some more comment to give an overview how function constants are meant to function for DX12. Maybe even given an example of a shader option, how it is patched in shader pipeline and how it is utilized at runtime.